Helper/Requests.php

```diff
@@ -381,18 +365,13 @@ public function buildRedirectData($payment, $request = [])
      */
     public function buildRecurringData(int $storeId, $payment, $request = [])
     {
-
-        $request['shopperInteraction'] = 'Ecommerce';
-
-        // TODO refactor to set the shopperInteraction only this place (only one place) because now it's going to be
-        // overriden in the OneclickAuthorizationDataBuilder
-
         $enableOneclick = $this->adyenHelper->getAdyenAbstractConfigData('enable_oneclick', $storeId);
         $enableVault = $this->adyenHelper->isCreditCardVaultEnabled();
         $storedPaymentMethodsEnabled = $this->adyenHelper->getAdyenOneclickConfigData('active', $storeId);
 
         // TODO Remove it in version 7
-        if ($payment->getAdditionalInformation(AdyenCcDataAssignObserver::STORE_CC)) {
+        if ($payment->getAdditionalInformation(AdyenCcDataAssignObserver::STORE_CC) ||
+            $payment->getMethodInstance()->getCode() == AdyenPayByLinkConfigProvider::CODE) {
             $request['storePaymentMethod'] = true;
         }
```

Comment:
Should we remove this now that we're on v7 according to the TODO? Did it mean the if check or the. `storePaymentMethod ` object?

Reply:
I believe it is indeed not needed anymore since we are using the `storePaymentMethod` field from the checkout component in the `state.data`. Just to make sure that it's indeed not used after removing it could you please verify that the same field is used from the `state.data` object?

Reply:
Ok I changed it to this https://github.com/Adyen/adyen-magento2/pull/1090/commits/35d4f485ce79b37b3df38efea8fb6cfb6e6b8cd6

Because we need to send the `storePaymentMethod` flag for PMs with and without component I first check the state.data value and only use the M2 config if there's no value set (e.g. no component like with PBL).